kernel/uid_gid.c

```diff
@@ -6,32 +6,89 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include <myst/eraise.h>
 #include <myst/kernel.h>
+#include <myst/mount.h>
 #include <myst/syscall.h>
 #include <myst/thread.h>
 
-/* Currently only supports a single mapping which is typically from enclave root
- * identity to the host application identity when executed. */
-static myst_host_enc_id_mapping host_enc_id_mapping;
+static myst_host_enc_uid_mapping uid_mappings[MAX_ID_MAPPINGS];
+static int num_uid_mappings;
+static myst_host_enc_gid_mapping gid_mappings[MAX_ID_MAPPINGS];
+static int num_gid_mappings;
 
-void myst_set_host_uid_gid_mapping(myst_host_enc_id_mapping id_mapping)
+void myst_set_host_uid_gid_mappings(
```

Comment:
Rename to `myst_copy_host_uid_gid_mappings`

Reply:
Fixed.